src/services/castleads.pas

```diff
@@ -300,7 +321,7 @@ procedure TAds.TAdNetworkHandler.ShowFullScreenAd(const AdType: TFullScreenAdTyp
 begin
   { if the network doesn't support showing full-screen ads, pretend it's shown,
     in case user code waits for OnFullScreenAdClosed. }
-  FullScreenAdClosed(false);
+  FullScreenAdClosed(wsUnknownError);
```

Comment:
Maybe make it `wsAdTypeUnsupported` or something like this. I feel uneasy calling `wsUnknownError` at this point, since we actually know what is the reason :)